xtr/src/generator.rs

```diff
@@ -75,10 +75,21 @@ msgstr ""
                 writeln!(output, "#. {}", c)?;
             }
         }
-        if !m.locations.is_empty() {
+        if !m.locations.is_empty() && (output_details.add_location != AddLocation::Never) {
             write!(output, "#:")?;
             for l in &m.locations {
-                write!(output, " {}:{}", l.file.to_string_lossy(), l.line)?;
+                match output_details.add_location {
+                    AddLocation::Full => {
+                        write!(output, " {}:{}", l.file.to_string_lossy(), l.line)?;
+                    }
+                    AddLocation::File => {
+                        write!(output, " {}", l.file.to_string_lossy())?;
+                    }
+                    _ => panic!(format!(
```

Comment:
i think you don't need to use format! here.  panic itself should support formating.